x/wasm-storage/keeper/filter.go

```diff
@@ -39,3 +54,55 @@ func ApplyFilter(filter []byte, reveals []RevealBody) ([]int, bool, error) {
 		return outliers, false, nil
 	}
 }
+
+// filterMode takes in a list of reveals, and a json path to get the
+// filtering data from the RevealBody->reveal.
+// The function returns a list of int where value of each element can be
+// either 0 or 1. At index i Value 1 means RevealBody[i] contains an
+// outlier. If a reveal have MORE THAN 2/3 frequency, than it's not an
+// outlier.
+//
+// The function also returns if a consensus is reached. Consensus can be
+// reached in two ways.
+// 1. More than 1/3 of the reveals are corrupted. i.e. has non-zero exit code.
+// 2. More than 2/3 of the data is the same.
+func filterMode(dataPath string, reveals []RevealBody) ([]int, bool) {
+	var maxFreq, nonZeroExitCode int
+	freq := make(map[string]int, len(reveals))
+	revealVals := make([]string, 0, len(reveals))
+
+	for _, r := range reveals {
+		val := gjson.Get(r.Reveal, dataPath).String()
```

Comment:
Is there reason why we use `gjson` package here instead of unmarshalling JSON the standard way?

Reply:
Yes because the json is unstructured and can get deep. Check the unit tests for some example.